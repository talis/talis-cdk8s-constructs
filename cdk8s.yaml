language: typescript
imports:
  - k8s@1.30.0
