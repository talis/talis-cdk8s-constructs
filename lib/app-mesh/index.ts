export * from "./ingress-gateway-nlb";
export * from "./app-service";
