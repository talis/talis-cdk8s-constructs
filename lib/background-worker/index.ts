export * from "./background-worker-props";
export * from "./background-worker";
