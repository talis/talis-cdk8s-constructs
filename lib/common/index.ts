export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
