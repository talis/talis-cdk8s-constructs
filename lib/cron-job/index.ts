export * from "./cron-job-props";
export * from "./cron-job";
