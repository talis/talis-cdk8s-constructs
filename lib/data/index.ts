export * from "./config-map";
export * from "./secret";
export * from "./factory";
