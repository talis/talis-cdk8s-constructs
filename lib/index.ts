export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
