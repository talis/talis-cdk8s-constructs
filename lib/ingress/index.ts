export * from "./ingress-props";
export * from "./ingress";
