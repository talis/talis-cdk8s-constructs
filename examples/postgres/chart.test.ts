import { PostgresChart } from "./chart";
import { Testing } from "cdk8s";
import { TalisShortRegion, TalisDeploymentEnvironment } from "../../lib";

describe("Postgres example", () => {
  beforeEach(() => {
    vi.resetModules();
  });

  test("Snapshot", () => {
    const app = Testing.app();
    const chart = new PostgresChart(app, {
      environment: TalisDeploymentEnvironment.DEVELOPMENT,
      region: TalisShortRegion.LOCAL,
      watermark: "test",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
