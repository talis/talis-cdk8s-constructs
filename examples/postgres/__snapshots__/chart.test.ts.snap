// Vitest Snapshot v1

exports[`Postgres example > Snapshot 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": {
      "labels": {
        "app": "example-postgres-app",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "example-postgres-app-development-local",
      },
      "name": "example-postgres-app-test",
      "namespace": "example-postgres-app-test",
    },
  },
  {
    "apiVersion": "v1",
    "data": {
      "create-schema.sql": "CREATE TABLE sample(
          id uuid,
          name varchar(64),
          creation_date date
        );",
    },
    "kind": "ConfigMap",
    "metadata": {
      "labels": {
        "app": "example-postgres-app",
        "environment": "development",
        "managed-by": "cdk8s",
        "prunable": "true",
        "region": "local",
        "service": "example-postgres-app-development-local",
      },
      "name": "postgres-init-27d7dggmb6",
      "namespace": "example-postgres-app-test",
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "labels": {
        "app": "example-postgres-app",
        "environment": "development",
        "instance": "postgres-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "9.5",
        "role": "postgres",
        "service": "example-postgres-app-development-local",
      },
      "name": "postgres-example",
      "namespace": "example-postgres-app-test",
    },
    "spec": {
      "clusterIP": "None",
      "ports": [
        {
          "port": 5432,
          "protocol": "TCP",
        },
      ],
      "selector": {
        "app": "example-postgres-app",
        "instance": "postgres-example",
        "role": "postgres",
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": {
      "labels": {
        "app": "example-postgres-app",
        "environment": "development",
        "instance": "postgres-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "9.5",
        "role": "postgres",
        "service": "example-postgres-app-development-local",
      },
      "name": "postgres-example-sts",
      "namespace": "example-postgres-app-test",
    },
    "spec": {
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "app": "example-postgres-app",
          "instance": "postgres-example",
          "role": "postgres",
        },
      },
      "serviceName": "postgres-example",
      "template": {
        "metadata": {
          "labels": {
            "app": "example-postgres-app",
            "environment": "development",
            "instance": "postgres-example",
            "managed-by": "cdk8s",
            "region": "local",
            "release": "9.5",
            "role": "postgres",
            "service": "example-postgres-app-development-local",
          },
        },
        "spec": {
          "containers": [
            {
              "env": [
                {
                  "name": "POSTGRES_PASSWORD",
                  "value": "secret",
                },
                {
                  "name": "POSTGRES_USER",
                  "value": "appuser",
                },
                {
                  "name": "POSTGRES_DB",
                  "value": "mypostgresdb",
                },
              ],
              "image": "postgres:9.5",
              "livenessProbe": {
                "exec": {
                  "command": [
                    "pg_isready",
                  ],
                },
                "failureThreshold": 3,
                "initialDelaySeconds": 30,
                "periodSeconds": 10,
                "successThreshold": 1,
                "timeoutSeconds": 5,
              },
              "name": "postgres",
              "ports": [
                {
                  "containerPort": 5432,
                },
              ],
              "readinessProbe": {
                "exec": {
                  "command": [
                    "pg_isready",
                  ],
                },
                "failureThreshold": 3,
                "initialDelaySeconds": 30,
                "periodSeconds": 10,
                "successThreshold": 1,
                "timeoutSeconds": 5,
              },
              "resources": {
                "limits": {
                  "cpu": "100m",
                  "memory": "500Mi",
                },
              },
              "volumeMounts": [
                {
                  "mountPath": "/var/lib/postgresql/data",
                  "name": "postgres-data",
                  "subPath": "postgres",
                },
                {
                  "mountPath": "/docker-entrypoint-initdb.d",
                  "name": "initdb",
                },
              ],
            },
          ],
          "volumes": [
            {
              "configMap": {
                "name": "postgres-init-27d7dggmb6",
              },
              "name": "initdb",
            },
          ],
        },
      },
      "volumeClaimTemplates": [
        {
          "metadata": {
            "name": "postgres-data",
          },
          "spec": {
            "accessModes": [
              "ReadWriteOnce",
            ],
            "resources": {
              "requests": {
                "storage": "20Gi",
              },
            },
            "storageClassName": "general-purpose-delete",
          },
        },
      ],
    },
  },
]
`;
